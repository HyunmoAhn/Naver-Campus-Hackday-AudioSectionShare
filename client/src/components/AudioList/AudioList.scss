.AudioList {
  border: 1px solid green;
  border-radius: 15px;
  margin: 15px;
  padding: 10px;
}