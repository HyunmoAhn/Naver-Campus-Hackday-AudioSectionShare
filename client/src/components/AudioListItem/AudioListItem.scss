.AudioListItem {
  border-bottom: 2px solid green;
  padding: 2px;
  text-align: center;
  overflow: hidden;
}

.AudioListItem__title {
  width: 90%;
  float: left;
  white-space: nowrap;
  overflow: scroll;
}

.AudioListItem__count {
  width: 10%;
  float: right;
  border-left: 3px solid green;
}