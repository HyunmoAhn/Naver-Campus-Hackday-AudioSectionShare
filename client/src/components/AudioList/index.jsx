import React from 'react';
import PropTypes from 'prop-types';
import AudioListItem from '../AudioListItem';
import './AudioList.scss';

const propTypes = {
	list: PropTypes.array,
	router: PropTypes.object,
	onListFetch: PropTypes.func,
	onListInfo: PropTypes.func,
};

const defaultProps = {
	list: [],
	router: {},
	onListFetch() {},
	onListInfo() {},
};

class AudioList extends React.Component {
	componentDidMount() {
		this.props.onListFetch();
	}
	render() {
		return (
			<div className="AudioList">
				{this.props.list.map((item, index) =>
					<AudioListItem
						key={index}
						router={this.props.router}
					  item={item}
						onListInfo={this.props.onListInfo}
					/>)
				}
			</div>
		);
	}
}

AudioList.propTypes = propTypes;
AudioList.defaultProps = defaultProps;

export default AudioList;