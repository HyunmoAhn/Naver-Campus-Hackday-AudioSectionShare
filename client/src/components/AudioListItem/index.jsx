import React from 'react';
import PropTypes from 'prop-types';
import './AudioListItem.scss';

const propTypes = {
	item: PropTypes.object.isRequired,
	router: PropTypes.object,
	onListInfo: PropTypes.func,
};

class AudioListItem extends React.Component {
	render() {
		const { count, id, title } = this.props.item;
		return (
			<div
				className="AudioListItem"
			  onClick={() => {
					this.props.onListInfo(id);
				  this.props.router.push(`?id=${id}&isList=true`);
			  }}
			>
				<span className="AudioListItem__title">
					{title}
				</span>
				<span className="AudioListItem__count">
					{count}
				</span>
			</div>
		)
	}
}

AudioListItem.propTypes = propTypes;

export default AudioListItem;