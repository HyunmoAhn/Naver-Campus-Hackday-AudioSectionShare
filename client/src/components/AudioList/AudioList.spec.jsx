import React from 'react';
import { shallow, mount } from 'enzyme';

import AudioList from './';
import AudioListItem from '../AudioListItem';

describe('<AudioList />', () => {
	const mockList = [
		{
			title: 'mock-title-1',
			id: 'mock-id-1',
			count: 5,
		},
		{
			title: 'mock-title-2',
			id: 'mock-id-2',
			count: 10,
		},
	];

	it('should match to snapshot when render default', () => {
		const wrapper = shallow(<AudioList list={mockList} />);

		expect(wrapper).toMatchSnapshot();
	});

	it('should call props.onListFetch when componentDidMount', () => {
		const onListFetch = jest.fn();
		const wrapper = mount(<AudioList list={mockList} onListFetch={onListFetch} />);

		expect(onListFetch.mock.calls.length).toBe(1);
	});

	it('should call collect when props.onListInfo of AudioListItem', () => {
		const onListInfo = jest.fn();
		const onSetFalseSectionLoop = jest.fn();
		const wrapper = shallow(<AudioList
			list={mockList}
		  onListInfo={onListInfo}
		  onSetFalseSectionLoop={onSetFalseSectionLoop}
		/>);
		const audioListItem = wrapper.find(AudioListItem).at(0);

		audioListItem.props().onListInfo(audioListItem.props().item.id);
		expect(onListInfo.mock.calls.length).toBe(1);
		expect(onListInfo.mock.calls[0]).toEqual([audioListItem.props().item.id]);
		expect(onSetFalseSectionLoop.mock.calls.length).toBe(1);
	});
});