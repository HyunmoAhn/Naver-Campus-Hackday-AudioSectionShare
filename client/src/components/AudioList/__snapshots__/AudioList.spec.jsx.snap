// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AudioList /> should match to snapshot when render default 1`] = `
<div
  className="AudioList"
>
  <AudioListItem
    item={
      Object {
        "count": 5,
        "id": "mock-id-1",
        "title": "mock-title-1",
      }
    }
    key="0"
    onListInfo={[Function]}
    router={Object {}}
  />
  <AudioListItem
    item={
      Object {
        "count": 10,
        "id": "mock-id-2",
        "title": "mock-title-2",
      }
    }
    key="1"
    onListInfo={[Function]}
    router={Object {}}
  />
</div>
`;
